lib/Wikia/src/Service/User/Attributes/UserAttributes.php

```diff
@@ -64,38 +64,77 @@ private function loadAttributes( $userId ) {
 	 * @param string $userId
 	 * @param Attribute $attribute
 	 */
-	public function setAttribute( $userId, $attribute ) {
+	public function setAttribute( $userId, Attribute $attribute ) {
 		if ( $this->isAnonUser( $userId ) ) {
 			return;
 		}
 
-		$this->loadAttributes( $userId );
-		$this->setAttributeInService( $userId, $attribute );
-		$this->setAttributeInCache( $userId, $attribute );
+		if ( $this->attributeValueAlreadySet( $userId, $attribute ) ) {
+			return;
+		}
 
+		$this->setAttributeInCache( $userId, $attribute );
+		$this->setAttributeInService( $userId, $attribute );
```

Comment:
Not that I can recall specifically. Setting the service first makes sense to me. I'll unswitch.